src/sentry/utils/pytest/sentry.py

```diff
@@ -65,6 +66,12 @@ def pytest_configure(config):
                     "HOST": "127.0.0.1",
                 }
             )
+            # Unfortunately, e2e tests use a threaded server and access endpoints concurrently.
+            # For hybrid cloud testing, we need to simulate a split silo environment in tests,
+            # which means swapping out the database user contexts during test runs.  But we need
+            # this to be thread safe alongside the django.db.connections thread.local managing connections.
+            settings.DATABASES = ThreadLocalDict(settings.DATABASES)
```

Comment:
This also can be removed from the PR -- it was an experiment I was using to simulate per region data access by swapping out connections / users as endpoints are called.  But it was never finished and I think I can add this separately later.